src/buffer/out/TextAttribute.hpp

```diff
@@ -161,6 +164,13 @@ class TextAttribute final
     {
         return WI_IsAnyFlagSet(_wAttrLegacy, COMMON_LVB_GRID_HORIZONTAL | COMMON_LVB_GRID_LVERTICAL | COMMON_LVB_GRID_RVERTICAL | COMMON_LVB_UNDERSCORE);
     }
+    constexpr bool HasAnyExtendedAttributes() const noexcept
+    {
+        return GetExtendedAttributes() != ExtendedAttributes::Normal ||
+               !IsAnyGridLineEnabled() ||
+               GetHyperlinkId() != 0 ||
+               IsReverseVideo();

```

Comment:
BTW `!IsAnyGridLineEnabled() || IsReverseVideo()` is the same as `_wAttrLegacy != 0`, _if_ you fix the constructor to not use `META_ATTRS`. Coincidentally it shouldn't use that anyways, but rather:
```cpp
COMMON_LVB_GRID_HORIZONTAL | COMMON_LVB_GRID_LVERTICAL | COMMON_LVB_GRID_RVERTICAL | COMMON_LVB_REVERSE_VIDEO | COMMON_LVB_UNDERSCORE
```
(This is because `TextAttribute` doesn't store DBCS attributes anymore.)